src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -688,6 +712,23 @@ private void RenderBufferedAttributeValueAccessor(CSharpCodeWriter writer)
             }
         }
 
+        private static bool ContainsChildContent(IList<Chunk> children)
+        {
+            foreach (var child in children)
+            {
+                if (!(child is ParentChunk) || child is TagHelperChunk)
```

Comment:
This is why I was suggesting that we make `ParentChunk` abstract, and specialize other nodes, so that you can generically traverse it :+1: 

Not relevant for this PR, just making a point.